pyanaconda/ilog.py

```diff
@@ -0,0 +1,40 @@
+#
+# ilog.py: Path based names logger provider.
+#
+# Copyright (C) 2017  Red Hat, Inc.  All rights reserved.
+#
+# This program is free software; you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation; either version 2 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <http://www.gnu.org/licenses/>.
+#
+
+# Q: Why do we have a separate module just for this
+#
+# A: To avoid import cycles. The anaconda_log module would be a natural
+#    place for this function, but it unfrotunatelly imports the flags module.
+#    And the flags module would have to import anaconda_log to obtain the
+#    get_logger() function, resulting in an import cycle.
+#    We should be able to avoid this be placing the get_logger() function
+#    in its own module that does not import any Anaconda modules.
+
+import logging
+
+def get_logger(module_name):
```

Comment:
I guess that this could be done also in the `__init__` method of the custom `Logger` class, that could be set as default with `logging.setLoggerClass`. Then we could get loggers as:

```
import logging
log = logging.getLogger(__name__)
```
Have you considered that?